### PR TITLE
Fixing bug in setter on Proxy object construction

### LINKED ISSUE #425: GaudiPython application broken in 6.1.18 release
In the latest release the GaudiPython application is broken and ignores the project that is specified.
Here is a trivial example:
```
[02:32:54]
Ganga In [1]: j=Job(application=GaudiPython(project='LHCb',version='v39r0'),backend=Interactive())
j.s
[02:33:02]
Ganga In [2]: j.submit()
Ganga.GPIDev.Lib.Job               : INFO     submitting job 88
Ganga.GPIDev.Lib.Job               : INFO     job 88 status changed to "submitting"
Ganga.GangaGaudi.Lib.Applications  : INFO     Job 88: Preparing GaudiPython application.
Ganga.GangaLHCb.Lib.Applications   : WARNING  No script defined. Will use a default script which is probably not what you want.
Ganga.GPIDev.Lib.Job               : INFO     Preparing subjobs
Ganga.GPIDev.Adapters              : INFO     submitting job 88 to Interactive backend
Ganga.Lib.Interactive              : INFO     Starting job 88
Ganga.GPIDev.Lib.Job               : INFO     job 88 status changed to "submitted"
WARNING : using environment variable CMTSITE for cmtsite
WARNING : using environment variable User_release_area for userarea
WARNING : using environment variable VO_LHCB_SW_DIR for sharedarea
ALWAYS  : Configuring LbScripts v8r5p3 from /cvmfs/lhcb.cern.ch/lib/lhcb/LBSCRIPTS/LBSCRIPTS_v8r5p3
Configuring LbScripts v8r5p3 from /cvmfs/lhcb.cern.ch/lib/lhcb/LBSCRIPTS/LBSCRIPTS_v8r5p3
********************************************************************************
*                         ---- LHCb Login v8r5p3 ----                          *
*      Building with gcc49 on slc6 x86_64 system (x86_64-slc6-gcc49-opt)       *
********************************************************************************
 --- User_release_area is set to /home/lupton/cmtuser
 --- LHCBPROJECTPATH is set to:
    /cvmfs/lhcb.cern.ch/lib/lhcb
    /cvmfs/lhcb.cern.ch/lib/lcg/releases
    /cvmfs/lhcb.cern.ch/lib/lcg/app/releases
    /cvmfs/lhcb.cern.ch/lib/lcg/external
--------------------------------------------------------------------------------
Using CMTPROJECTPATH = '/cvmfs/lhcb.cern.ch/lib/lhcb:/cvmfs/lhcb.cern.ch/lib/lcg/releases:/cvmfs/lhcb.cern.ch/lib/lcg/app/releases:/cvmfs/lhcb.cern.ch/lib/lcg/external'
Environment for LbScripts v8r5p3 ready.
(Compat v1r19 from /cvmfs/lhcb.cern.ch/lib/lhcb/COMPAT/COMPAT_v1r19,
 LbScripts v8r5p3 from /cvmfs/lhcb.cern.ch/lib/lhcb/LBSCRIPTS/LBSCRIPTS_v8r5p3,
 LCGCMT 84 from /cvmfs/lhcb.cern.ch/lib/lcg/releases/LCGCMT/LCGCMT_84,
 Compat v1r19 from /cvmfs/lhcb.cern.ch/lib/lhcb/COMPAT/COMPAT_v1r19)
Configuring DaVinci v39r0 from /cvmfs/lhcb.cern.ch/lib/lhcb/DAVINCI/DAVINCI_v39r0
```
Where the last line shows that it is ignoring the specified application (LHCb) and using DaVinci instead, although it is using the correct version.
The same problem is present in the develop branch, which I happened to have ready to test.

### PR DESCRIPTION
**Pull request by rob-c**
This fixes a bug in the Proxy constructor which doesn't respect the functionality which is in the ```...filter__set__``` method.

This is required for the IGangaFile objects to behave how I'm expecting them to and so should be go in as it fixes a bug which is easy to reproduce in that setting an ```.lfn``` of a ```DiracFile``` fails to modify the ```namePattern``` accordingly as would naively be expected.

Yes if we manage to get the filter system to go away this can be reverted, but for now this is a bug and needs to be fixed.

This should also fix #425 although this is just an afterthought and I've done exactly 0% testing of this particular problem.

I'm keen to get this in asap so I can carry on with LSST work.

Fixes #425

### CODE REVIEW

File: python/Ganga/GPIDev/Base/Proxy.py [hunk 1]
```diff
@@ -788,9 +788,15 @@ def _init(self, *args, **kwds):
                 val = stripProxy(getattr(self, key))
                 if isinstance(val, GangaObject):
                     val._auto__init__()
-                    instance.setNodeAttribute(key, stripProxy(val))
+                    if hasattr(instance, '_attribute_filter__set__'):
```
Comment (milliams) [inline python/Ganga/GPIDev/Base/Proxy.py]: Every `GangaObject` has `_attribute_filter__set__` since it's defined on the base class.

**Comment by rob-c** [comment]
Any objections to merging this?

**Comment by milliams** [comment]
I think this makes sense. It seems like it's duplicating some of the logic from `ProxyDataDescriptor.__set__`. They could perhaps instead be replaced with:
```python
self.setattr(key, val)
```
which will make sure that the same code is run over the attributes passed in via constructor arguments as with attribute assignment.

**Comment by rob-c** [comment]
@milliams That is potentially much more invasive PR and I'm keen to get this fixed asap. There was some code which was more specialized in the ```def _init``` but in time this is looking more akin to the ```__set__``` function which I'm glad about.
or, do you simply mean replace ```setNodeAttr...``` with ```setattr``` in these 4 lines and check it does what we expect?

**Comment by milliams** [comment]
Sorry, I was perhaps a bit short in my explanation. I do indeed mean the latter. Replace what is currently:
```python
instance.setNodeAttribute(key, stripProxy(val))
```
with
```python
setattr(self, key, val)
```
There's no good reason here for us to be purposefully circumventing the `Descriptor.__set__` is there? Unless I'm missing something.

**Comment by rob-c** [comment]
I'll check this fixes the original issue and amend as appropriate so I can do some other work.

**Comment by rob-c** [comment]
OK, setattr from elsewhere in the file does what I want. I'm going to merge this once the tests finish

**Pull request merged**